File: frontend/types/materials.ts
export type MaterialSource = "youtube" | "article" | "website";

export interface Material {
  id: string;
  title: string;
  source: MaterialSource;
  url: string;
  description?: string;
  thumbnail?: string;
}

export interface AIMaterialsResponse {
  materials: Material[];
  recommendation: string;
}

File: backend/app/schemas/materials.py
from __future__ import annotations

from pydantic import BaseModel, HttpUrl


class MaterialSearchResult(BaseModel):
    title: str
    url: HttpUrl
    source: str
    description: str | None = None
    thumbnail_url: HttpUrl | None = None


class AIMaterialsResponse(BaseModel):
    materials: list[MaterialSearchResult]
    recommendation: str = ""
